x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/index_data_visualizer_view/index_data_visualizer_view.tsx

```diff
@@ -306,12 +307,14 @@ export const IndexDataVisualizerView: FC<IndexDataVisualizerViewProps> = (dataVi
 
   useEffect(() => {
     return () => {
-      // When navigating away from the index pattern
-      // Reset all previously set filters
-      // to make sure new page doesn't have unrelated filters
-      data.query.filterManager.removeAll();
+      // We want to clear all filters that have not been pinned globally
+      // when navigating to other pages
+      data.query.filterManager
+        .getFilters()
+        .filter((f) => f.$state?.store === FilterStateStore.APP_STATE)
+        .forEach((f) => data.query.filterManager.removeFilter(f));
     };
-  }, [currentDataView.id, data.query.filterManager]);
+  }, [data.query.filterManager]);
```

Comment:
spotted the same effect `explain_log_rate_spikes_page.tsx`. Shall we create a shared hook `useClearFiltersOnLeave`?

Reply:
As discussed in the meeting, this will be done in a follow up. I marked it with a todo for now.